### ISSUE TITLE
feature request: network create does not have "--label"

### ISSUE THREAD

**Issue by medyagh** [containers/podman#5376]
in minikube we recently added podman as a driver, however we can not create a network with a label like for docker.

the reason for creating a network with a label is, minikube delete should be able to clean up after itself and delete all the created containers and networks.

```
 podman network create podman1

Flags:
      --disable-dns      disable dns plugin
  -d, --driver string    driver to manage the network (default "bridge")
      --gateway ip       IPv4 or IPv6 gateway for the subnet
      --internal         restrict external access from this network
      --ip-range ipNet   allocate container IP from range
      --macvlan string   create a Macvlan connection based on this device
      --subnet ipNet     subnet in CIDR format
```

is there any chance to match docker on this feature ?

```
Usage:	docker network create [OPTIONS] NETWORK

Create a network

Options:
      --attachable           Enable manual container attachment
      --aux-address map      Auxiliary IPv4 or IPv6 addresses used by Network driver (default map[])
      --config-from string   The network from which copying the configuration
      --config-only          Create a configuration only network
  -d, --driver string        Driver to manage the Network (default "bridge")
      --gateway strings      IPv4 or IPv6 Gateway for the master subnet
      --ingress              Create swarm routing-mesh network
      --internal             Restrict external access to the network
      --ip-range strings     Allocate container ip from a sub-range
      --ipam-driver string   IP Address Management Driver (default "default")
      --ipam-opt map         Set IPAM driver specific options (default map[])
      --ipv6                 Enable IPv6 networking
      --label list           Set metadata on a network
  -o, --opt map              Set driver specific options (default map[])
      --scope string         Control the network's scope
      --subnet strings       Subnet in CIDR format that represents a network segment
```

**Comment by baude** [comment]
is it not sufficient to podman network rm -f <name_of_network>?

**Comment by mheon** [comment]
Can you clarify the use case here a bit? Are you using labels and filtering with `podman network ls --filter` to identify what networks need to be removed?

**Comment by rhatdan** [comment]
@kunalkushwaha Interested in working on this? @sujil02 ?

**Comment by hitchhooker** [comment]
> Can you clarify the use case here a bit? Are you using labels and filtering with `podman network ls --filter` to identify what networks need to be removed?
`podman network ls --filter ` does there even exist such flag as --filter for network? I keep receiving unknown flag error when trying to replace docker with podman for matrix-docker-ansible-deploy. 

`podman network ls -q --filter='name=^{{ matrix_podman_network }}$'
Error: unknown flag: --filter`

**Comment by rhatdan** [comment]
@ParkerVR PTAL @baude Will help you out.

**Comment by mheon** [comment]
@rhatdan This one isn't going to be easy - basically going to need a complete rewrite of our network handling code.

**Comment by dougsland** [comment]
> > Can you clarify the use case here a bit? Are you using labels and filtering with `podman network ls --filter` to identify what networks need to be removed?
> 
> `podman network ls --filter ` does there even exist such flag as --filter for network? I keep receiving unknown flag error when trying to replace docker with podman for matrix-docker-ansible-deploy.
> 
> `podman network ls -q --filter='name=^{{ matrix_podman_network }}$'`
> `Error: unknown flag: --filter`

@hitchhooker looks like it works for me.
```
# ./bin/podman network ls --filter name=podman 
NAME    VERSION  PLUGINS
podman  0.4.0    bridge,portmap,firewall,tuning
```

```
 ./bin/podman version
Version:      2.1.0-dev
API Version:  1
Go Version:   go1.13.6
Git Commit:   11e237bc3a6ab7ccb3a05ea5ffb1b6b046c1bac3-dirty
Built:        Sun Jun 21 07:01:57 2020
OS/Arch:      linux/amd64
```

**Comment by dougsland** [comment]
@mheon @rhatdan how can I help in this one?

**Comment by rhatdan** [comment]
Need @mheon advise on this one.

**Comment by mheon** [comment]
This basically needs a full rewrite of how we handle networking, to store networks we create inside Libpod's database instead of using CNI config files as the sole source of truth (or CNI needs to add a labels field to their configs, which given our past interactions with them on feature requests, seems highly unlikely). There are cards already created for this, I think @baude wants me on it next sprint.

**Comment by rhatdan** [comment]
I don't think we have made any progress on this

**Comment by medyagh** [comment]
> is it not sufficient to podman network rm -f <name_of_network>?

No, because I wouldn't know if that network was created by our tool (in this case minikube) or by the user manually. and minikube doesn't wanna accidentally delete users created networks.

we rely on a specific label we set on the network to mark them as created by minikube

**Comment by Luap99** [comment]
Fixed in #8505
